client/daemon/rpcserver/rpcserver.go

```diff
@@ -295,3 +310,237 @@ func (s *server) Download(ctx context.Context,
+		msg := fmt.Sprintf("import file failed: %v", err)
+		log.Error(msg)
+		return nil, errors.New(msg)
+	} else {
+		log.Info("import file succeeded")
+	}
+
+	// 3. Announce to scheduler asynchronously
+	go announceFunc()
+
+	return common.NewGrpcDfResult(base.Code_Success, "import file succeeded"), nil
+}
+
+func (s *server) ExportTask(ctx context.Context, req *dfdaemongrpc.ExportTaskRequest) (*base.GrpcDfResult, error) {
+	taskID := idgen.TaskID(req.Cid, req.UrlMeta)
+	log := logger.With("function", "ExportTask", "Cid", req.Cid, "Tag", req.UrlMeta.Tag, "taskID", taskID, "destination", req.Output)
+
+	log.Info("new export task request")
+	task := s.storageManager.FindCompletedTask(taskID)
+	if task == nil {
+		// If only use local cache and task doesn't exist, return error
+		if req.LocalOnly {
+			log.Info("task not found in local storage")
+			msg := fmt.Sprintf("task not found, taskID %s, cid %s", taskID, req.Cid)
+			return common.NewGrpcDfResult(base.Code_PeerTaskNotFound, msg), nil
+		}
+		log.Info("task not found, try from peers")
+		return s.exportFromPeers(ctx, log, req)
+	}
+	err := s.exportFromLocal(ctx, req, task.PeerID)
+	if err != nil {
+		log.Error("export from local failed: %s", err)
+		return nil, err
+	}
+	return common.NewGrpcDfResult(base.Code_Success, "task exported"), nil
+}
+
+func (s *server) isTaskCompleted(taskID string) bool {
+	return s.storageManager.FindCompletedTask(taskID) != nil
+}
+
+func (s *server) exportFromLocal(ctx context.Context, req *dfdaemongrpc.ExportTaskRequest, peerID string) error {
+	return s.storageManager.Store(ctx, &storage.StoreRequest{
+		CommonTaskRequest: storage.CommonTaskRequest{
+			PeerID:      peerID,
+			TaskID:      idgen.TaskID(req.Cid, req.UrlMeta),
+			Destination: req.Output,
+		},
+		StoreDataOnly: true,
+	})
+}
+
+func (s *server) exportFromPeers(ctx context.Context, log *logger.SugaredLoggerOnWith, req *dfdaemongrpc.ExportTaskRequest) (*base.GrpcDfResult, error) {
+	peerID := idgen.PeerID(s.peerHost.Ip)
+	taskID := idgen.TaskID(req.Cid, req.UrlMeta)
+
+	res, err := s.peerTaskManager.StatPeerTask(ctx, taskID)
+	if err != nil {
+		msg := fmt.Sprintf("failed to StatPeerTask from peers: %s", err)
+		log.Error(msg)
+		return nil, errors.New(msg)
+	}
+	if res.Code != base.Code_Success {
+		log.Infof("task not found in P2P network")
+		return common.NewGrpcDfResult(res.Code, res.Message), nil
+	}
+
+	var (
+		start     = time.Now()
+		drc       = make(chan *dfdaemongrpc.DownResult, 1)
+		errChan   = make(chan error, 10)
+		result    *dfdaemon.DownResult
+		downError error
+	)
+	downRequest := &dfdaemongrpc.DownRequest{
+		Url:               req.Cid,
+		Output:            req.Output,
+		Timeout:           req.Timeout,
+		Limit:             req.Limit,
+		DisableBackSource: true,
+		UrlMeta:           req.UrlMeta,
+		Pattern:           "",
+		Callsystem:        req.Callsystem,
+		Uid:               req.Uid,
+		Gid:               req.Gid,
+	}
+	once := new(sync.Once)
+	closeDrc := func() {
```

Comment:
In golang, channel will be auto closed and gc if there is no goroutine using it. In this case, `clouseDrc` can be removed to keep code simple.

Reply:
Fixed